tests/aws/services/apigateway/test_apigateway_lambda.py

```diff
@@ -832,3 +836,64 @@ def invoke_api(url):
         elif lambda_format_type == "wrong-format":
             assert response.status_code == 502
             assert response.json() == {"message": "Internal server error"}
+
+
+@markers.aws.validated
+@markers.only_on_amd64
+def test_lambda_rust_proxy_integration(
```

Comment:
awesome, I was worried about deleting the test above but didn't see it was migrated! 🥳

Reply:
@bentsku Does Rust do something weirdly specific here that is important for the API Gateway integration?

I'm primarily wondering about ARM compatibility. The API Gateway also has these two parametrized tests left in ext: https://github.com/localstack/localstack-ext/pull/2899
- tests/aws/services/apigateway/test_rest_apis.py::TestRestAPIs::test_custom_authorizer[NONE-GET]
- tests/aws/services/apigateway/test_rest_apis.py::TestRestAPIs::test_custom_authorizer[NONE-POST]
- tests/aws/services/apigateway/test_rest_apis.py::TestRestAPIs::test_custom_authorizer[CUSTOM-GET]
- tests/aws/services/apigateway/test_rest_apis.py::TestRestAPIs::test_custom_authorizer[CUSTOM-POST]
- tests/aws/services/apigateway/test_websocket_apis.py::TestWebSockets::test_websocket_api_with_http_integration[HTTP-False-True]
- tests/aws/services/apigateway/test_websocket_apis.py::TestWebSockets::test_websocket_api_with_http_integration[HTTP-False-False]

If we have a strong reason for using Rust rather here, we could consider adding it to an ARM-compatible cached build under the `lambda_/functions/common` as demonstrated here: `tests/aws/services/lambda_/functions/common/uncaughtexception/provided/Makefile`

Secondly, I'm not a big fan of packaged zip functions because they are black boxes: hard to understand and maintain.

----

Awesome work @cloutierMat 👏👏👏 🚀🚀🚀

Reply:
I believe it does, yes, most probably it's about the API Gateway context serialization/deserialization which must be pretty strict in Rust. Here's the PR introducing it. 

We could add it, yes! Not too sure what's the value of testing ARM here, as I believe it will behave strictly the same. 
Maybe @dfangl can chime in too, he was the author of the PR introducing it, but I think having Rust tests for API Gateway kinda makes sense.

Reply:
#6005 was the original PR

Reply:
> I believe it does, yes, most probably it's about the API Gateway context serialization/deserialization which must be pretty strict in Rust. Here's the PR introducing it.

Thanks for the clarification. I'm mainly curious about what makes this specific to Rust (beyond being a good regression example). Wouldn't snapshotting the context achieve the same?

> We could add it, yes! Not too sure what's the value of testing ARM here, as I believe it will behave strictly the same.

Valid assumption 👍  I think it's valuable to separate what fails on ARM via `@only_on_amd64` from "CI optimization skips" because we need a clearer picture of our ARM gaps (i.e., LS broken vs. test broken).

Reply:
> Thanks for the clarification. I'm mainly curious about what makes this specific to Rust (beyond being a good regression example). Wouldn't snapshotting the context achieve the same?

True, yes! I think Rust might be stricter than our snapshot regarding types, etc? However the PR was done before the snapshots were really into use. I think having at least one APIGW <-> lambda test with a strongly typed language shouldn't hurt just as a sanity check, but I agree with the second point about separating the markers. Any idea how we could go about that?

Reply:
Alright, two suggestions then:
1. **Snapshots vs. Rust**: Use snapshot testing wherever possible and add a comment to clarify the need for a Rust Lambda. Snapshots also gives more actionable errors compared to some random Rust serialization message.
2. **ARM skips**: Only use the existing Markers (e.g., `@only_on_amd`) for flagging broken LS functionality. We can temporarily use it for "broken" tests with an appropriate comment. Adding a `reason` field could be helpful. I would ignore pre-mature CI optimization for now.

Reply:
Thanks for the input @joe4dev. Based on it and given the scope of this ticket, I will make the following changes.

- Change the assert for a snapshot in the current Rust test
- Add comment to explain it's existence
- Remove the `only_on_amd` tag

There are already tests making snapshots of the lambda event context ie: `tests.aws.services.apigateway.test_apigateway_lambda.test_lambda_aws_proxy_integration` and adding new tests falls out of the current work scope: Remove `aws.unknown` tags.

Reply:
Chiming in here, as this is my test: it predates the snapshotting, so I am in principle fine with replacing it. However, I think it still has it's validity, as there are still some shortcomings in the snapshots (like types after snapshot transforms, date precision if opted out), where a test in one or multiple statically typed languages makes sense.

I would still advocate for having at least one test per payload Type in some statically typed languages, with the specific payload Type.

Reply:
Wouldn't removing the `only_amd_64` flag make the test fail now, as the function must be compiled for AMD64, hence the flag? 
This is the build command:
`cargo build --release --target x86_64-unknown-linux-musl`

I think we could leave it for now, as it's not exactly the scope of this PR, but leave a TODO on the test to address this in a future iteration to make it compatible for both AMD64 and ARM?

Reply:
Good catch! Will do!